samza-core/src/main/java/org/apache/samza/config/FileSystemCheckpointManagerConfig.java

```diff
@@ -16,16 +16,22 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+package org.apache.samza.config;
 
-package org.apache.samza.config
+import java.util.Optional;
 
-object FileSystemCheckpointManagerConfig {
-  // file system checkpoint manager config constants
-  val CHECKPOINT_MANAGER_ROOT = "task.checkpoint.path" // system name to use when sending offset checkpoints
 
-  implicit def Config2FSCP(config: Config) = new FileSystemCheckpointManagerConfig(config)
-}
+public class FileSystemCheckpointManagerConfig extends MapConfig {
+  /**
+   * Path on local file system where checkpoints should be stored.
+   */
+  private static final String CHECKPOINT_MANAGER_ROOT = "task.checkpoint.path";
```

Comment:
I asked myself this question too. I did it this way since then the test can help double check a change to a config key (which is kind of a public API) and make sure it is intentional.
Do you think it is still better to just use the variable in the test?

Reply:
I see your point. here is my opinion on this pattern
1. I am not sure if the string in the test is going to be considered as yet another place where the config is used and get simply renamed as opposed to insisting its importance to remind backward compatibility.
2. I also feel this new pattern targets only this config and we still need to apply this across our configs.

For these reasons, I'd prefer to reuse the variable.

In general, I agree maintaining backwards compatibility of configurations and treating them as public API isn't quite intuitive. One option to make it more explicit and intuitive is to move configurations to samza-api module.

Reply:
1. That's true. Requiring the renaming in the test is just an indirect way to remind about backwards compatibility. As you mentioned, there can be a more explicit way of maintaining backwards compatibility, but we can do that at a later time.
2. Good point as well. This pattern is inconsistent with existing tests.

I'll update the test to reuse the variable.